app/commands/community_video/retrieve.rb

```diff
@@ -0,0 +1,34 @@
+class CommunityVideo
+  class Retrieve
+    include Mandate
+
+    initialize_with :url
+
+    def call
+      raise InvalidCommunityVideoUrl unless url.present?
+
+      if youtube?
+        CommunityVideo::RetrieveFromYoutube.(url)
+      elsif vimeo?
+        CommunityVideo::RetrieveFromVimeo.(url)
+      else
+        raise InvalidCommunityVideoUrl
+      end
+    end
+
+    def youtube?
+      return true if url.starts_with?('https://youtu.be/')
+      return true if url.starts_with?('https://youtube.com')
+      return true if url.starts_with?(%r{https://[a-z]*\.?youtube.com/})
```

Comment:
Should this be:

```suggestion
      return true if url.starts_with?(%r{https://[a-z]+\.youtube.com/})
```

The case without a subdomain is already caught above.